### PR TITLE
[ci] Favor direct mage invocation on CI

### PR DESCRIPTION
**Pull request by andrewkroh**
## What does this PR do?

Favor direct mage invocation on CI.

This changes Jenkins and Travis to directly invoke mage where
possible instead of going through make. Some of the remaining
make don't yet have a mage equivalant (mainly crosscompile).

For Packetbeat this add Jenkins stages to test on darwin and Windows.

There were a few fixes I had to make related to these changes:

- Add some mage target dependencies to ensure fields and dashboards
are ready when tests use them.
- Swap the order of the Go imports for dev-tools/mage/targets/integtest
and unittest so that unit tests run before integ tests when running
the 'mage test' target.
- chown the shared Python venv that is in the root of the repo after
Dockerized integ tests exit to ensure there are no permissions issues
caused by root owned files.
- I found a few Python string encoding issues that caused tests failures.
I thought we fixed these during the Python 3 conversion, but something
here exposed a few that we didn't address.

One thing of note that I did not correct. Journalbeat has system tests
but they are not executed. The existing/old Makefile has SYSTEM_TESTS=false
so they are not executed. So when I switched it to mage I left it as
'mage goUnitTest' to avoid the Python when failed when I tried it.

## Why is it important?

We want to remove the older Makefile logic so we need to migrate towards mage.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by andrewkroh** [comment]
run tests

**Comment by andrewkroh** [comment]
run tests

**Comment by andrewkroh** [comment]
run tests

**Comment by andrewkroh** [comment]
run tests

**Comment by andrewkroh** [comment]
run tests

**Comment by andrewkroh** [comment]
run tests

File: Jenkinsfile [hunk 1]
```diff
@@ -387,19 +387,14 @@ pipeline {
...
-                makeTarget(context: "Libbeat oss Linux", target: "-C libbeat testsuite")
+                mageTarget(context: "Libbeat oss Linux", directory: "libbeat", target: "build test")
               }
             }
             stage('Libbeat crosscompile'){
               steps {
                 makeTarget(context: "Libbeat oss crosscompile", target: "-C libbeat crosscompile")
               }
             }
-            stage('Libbeat stress-tests'){
```
Comment (cachedout) [inline Jenkinsfile:398]: Could you comment more on this change, please? Is this being moved elsewhere and I'm just not seeing it or is it being removed entirely?

Comment (andrewkroh) [reply Jenkinsfile:398]: Good question. I forgot about this change. I should have mentioned it in the commit notes or reverted it. I found that this target never does anything, but I need to look into this more to figure out what it is supposed to be doing and fix it.

I'll revert this section for now.

**Comment by andrewkroh** [comment]
The failures in the build are caused by Kafka download failures while building the Docker image.

**Pull request merged**